back/app/services/project_copy_service.rb

```diff
@@ -33,25 +33,43 @@ def export(
     @project = project
     @template = { 'models' => {} }
 
+    shift_timestamps_and_local_copy = { shift_timestamps: shift_timestamps, local_copy: local_copy }
```

Comment:
If you store `local_copy` in an instance variable, you can avoid passing it in every method and just use it like that
```ruby
  def shift_timestamp(value, shift_timestamps)
    return if @local_copy
    value && (value + shift_timestamps.days)
  end
```

I assume, most of the diff will go away in this case.

Reply:
Well, I think we'll still need
```
'created_at' => @local_copy ? nil : shift_timestamp(image.created_at, shift_timestamps)&.iso8601,
'updated_at' => @local_copy ? nil : shift_timestamp(image.updated_at, shift_timestamps)&.iso8601
```
As without these conditionals, we would still set values for the 2 attributes, and thus ActiveRecord won't set them for us.

Reply:
But an instance var would avoid having to explicitly pas it to each method, yes.

Reply:
Hmm, clever. I'll do that!